httpsession/httpsession.go

```diff
@@ -0,0 +1,96 @@
+package httpsession
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/go-session-handler/config"
+	session "github.com/companieshouse/go-session-handler/session"
+	"github.com/companieshouse/go-session-handler/state"
+	"github.com/justinas/alice"
+)
+
+// ContextKey is a type to cast context keys
+type ContextKey string
+
+// ContextKeySession is the key used to fetch the session from the context
+var ContextKeySession = ContextKey("session")
+
+// Register will append an HTTP handler to an Alice chain, whereby the stored
+// session will be loaded and stored on the request context
+func Register(c alice.Chain) alice.Chain {
+	return c.Append(func(h http.Handler) http.Handler { return handler(h) })
+}
+
+// handler initialises a Store using config and cache structs, loads the
+// session, and stores it on the request context to access later
+func handler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		// Init all config
+		cfg := config.Get()
+
+		cache := state.NewCache(cfg.CacheServer, cfg.CacheDB, cfg.CachePassword)
+
+		s := state.NewStore(cache, cfg)
+
+		// Pull session ID from the cookie on the request
+		sessionID := getSessionIDFromRequest(cfg.CookieName, req)
+		var sessionData session.SessionData
+
+		// If session is stored, retrieve it from Redis
+		if sessionID != "" {
+
+			if err := s.Load(sessionID); err == nil {
+				sessionData = s.Data
+			} else {
+				log.ErrorR(req, err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}
+
+		ctx := context.WithValue(context.Background(), ContextKeySession, &sessionData)
+		req = req.WithContext(ctx)
+		h.ServeHTTP(w, req)
+
+		s.Data = sessionData
+
+		err := s.Store()
+		if err != nil {
+			log.ErrorR(req, err)
+		}
+
+		setSessionIDOnResponse(w, s, cfg)
```

Comment:
Personally I wouldn't necessarily factor this out into a separate function. It is called here, and only here, so other than for unit testing it does not offer much value, and the internal implementation is exposed through the function name (but that is difficult to "hide" when the function itself is just doing one simple thing as in this case).

Reply:
See above comment re: factoring code into separate functions.